internal/pkg/service/common/task/node.go

```diff
@@ -16,12 +16,12 @@ import (
 
 	"github.com/keboola/keboola-as-code/internal/pkg/idgenerator"
 	"github.com/keboola/keboola-as-code/internal/pkg/log"
-	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/schema"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/common/servicectx"
 	commonKey "github.com/keboola/keboola-as-code/internal/pkg/service/common/store/key"
 	commonModel "github.com/keboola/keboola-as-code/internal/pkg/service/common/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/store/schema"
```

Comment:
@michaljurecko Tady bych potřeboval poradit. `task/node` má závislost na `buffer/store/schema` protože ukládá task do db: https://github.com/keboola/keboola-as-code/blob/dd5e6bb9b2999d8f42831297143de4fd51eff0a2/internal/pkg/service/common/task/node.go#L173-L173
Tak jsem měl představu, že udělám `common/store/schema` interface s `Tasks()` metodou, Buffer bude mít svoje `Task` schema (s extra metodama `InReceiver()` a `InExport()` který se použijí jen tam) a Templates bude mít zase svoje schema. Ale narazil jsem na `buffer/dependencies/mocked` který mají metodu `Schema()` a tam se mi to nějak nedaří s tím interfacem abstrahovat. (Viz. tady nahoře v commitu.) Tak bych se nejdřív zeptal jestli vůbec mířím správným směrem a nemám to udělat nějak úplně jinak.

Reply:
Schema iba generuje `etcdop.Key` alebo `etcdop.Prefix`, mozes si ich vytvorit aj inde.

Spravil by som to takto:
- Do `task` pkg by som presunul:
   - TaskKey
   - Task model
- Pridal by som konfiguracnu moznost `TaskEtcdPrefix` by defaut `task`.
- Task prefix (ktory si ziskaval zo `schema`)  je potom uz iba `<TaskEtcdPrefix>/<TaskKey>`.

Reply:
(BTW: schema a model sa pravdepodobne neskor rozbiju na mensie casti `buffer/config`, `buffer/stats` ... a v jednom balicku bude aj schema, model aj logika .... bude to davat vacsi zmysel, ked bude menej kodu a pokope)

Reply:
🤔 `TaskKey` a `TaskID` už mám přesunuté (v prvním commitu: https://github.com/keboola/keboola-as-code/pull/1273/commits/63aea1d84b77c553ee6fe6fa67e413e7e5f2f3de) s tím problém nemám.

Problém je když jsem chtěl přesunout `service/buffer/task` do `common/buffer/task`, protože v `service/buffer/task/node` se dělá vložení tasku do etcd a je to závislý na `service/buffer/store/schema`: https://github.com/keboola/keboola-as-code/blob/97b78f408c172a6c760798ddcab5e98faa71f45d/internal/pkg/service/buffer/task/node.go#L173-L176

Reply:
no a `n.schema.Tasks().ByKey(task.TaskKey)` je to `<TaskEtcdPrefix>/<TaskKey>`.